addons/hr_holidays/models/hr_leave.py

```diff
@@ -489,33 +490,35 @@ def _compute_can_approve(self):
 
     @api.constrains('date_from', 'date_to', 'state', 'employee_id')
     def _check_date(self):
-        for holiday in self.filtered('employee_id'):
-            domain = [
-                ('date_from', '<', holiday.date_to),
-                ('date_to', '>', holiday.date_from),
-                ('employee_id', '=', holiday.employee_id.id),
-                ('id', '!=', holiday.id),
-                ('state', 'not in', ['cancel', 'refuse']),
-            ]
-            nholidays = self.search_count(domain)
-            if nholidays:
-                raise ValidationError(_('You can not set 2 times off that overlaps on the same day for the same employee.'))
+        employees = self.mapped('employee_id')
+        domains = [[
+            ('date_from', '<', holiday.date_to),
+            ('date_to', '>', holiday.date_from),
+            ('employee_id', '=', holiday.employee_id.id),
+            ('id', '!=', holiday.id),
+            ('state', 'not in', ['cancel', 'refuse']),
+        ] for holiday in self.filtered('employee_id')]
+        domain = expression.OR(domains)
+        if self.search_count(domain):
+            raise ValidationError(_('You can not set 2 times off that overlaps on the same day for the same employee.'))
 
     @api.constrains('state', 'number_of_days', 'holiday_status_id')
     def _check_holidays(self):
+        mapped_days = self.mapped('holiday_status_id').get_days(self.mapped('employee_id.id'))
```

Comment:
The semantics and logic seems odd. I'd flip the nesting to have the leave type on the outside, even if it means a bit more work for the backward compatibility thing. Or move the entire thing to hr.employees (e.g. `days_per_leave_type` or `totals_per_leave_type`) and the parameter would act as a filter on the leave types we want (possibly allowing for `None` -> every leave type? idunno). And then get_days might call that new method and do its munging internally without having to mess around any.

And the variable name is less than clear, it's "days per leave type per empoyee" so something along those lines?